src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -541,8 +553,8 @@ MulticopterAttitudeControl::control_attitude_rates(float dt)
 	_rates_prev = rates;
 	_rates_prev_filtered = rates_filtered;
 
-	/* update integral only if motors are providing enough thrust to be effective */
-	if (_thrust_sp > MIN_TAKEOFF_THRUST) {
+	/* update integral only if we are not landed */
+	if (!_vehicle_land_detected.maybe_landed && !_vehicle_land_detected.landed) {
```

Comment:
Why maybe_landed?

Reply:
So that during landing the integral does not load up. For instance when trying to land on uneven ground.
It might work without as well. Let's see what @MaEtUgR says.

Reply:
I think `maybe_landed` totally makes sense. ground_contact could in the worst case with wrong configuration have false positives in air and I think we don't want to reset the integral in that case. Landed is the last stage and a building up integral could prevent the detection because it requires no rotation movement for a certain timeout.

Here's a case where we can see again that it would make sense that if you are landed maybe_landed should also be true for better understanding and readability...

Reply:
> Here's again a case where we can see again that it would make sense that if you are landed maybe_landed should also be true for better understanding and readability

I think so too. It's also not clear from the uORB message, I had to look into the land detector implementation.